group-coordinator/src/main/java/org/apache/kafka/coordinator/group/modern/consumer/ConsumerGroup.java

```diff
@@ -342,6 +348,47 @@ private void removeStaticMember(ConsumerGroupMember oldMember) {
         }
     }
 
+    /**
+     * Update the regular expression.
+     *
+     * @param regex                         The regular expression.
+     * @param newResolvedRegularExpression  The regular expression's metadata.
+     */
+    public void updateRegularExpression(
```

Comment:
nit: how's updateResolvedRegularExpression and removeResolvedRegularExpression? and

"Update the resolved regular expression."